core/drivers/scmi-msg/entry.c

```diff
@@ -13,6 +18,65 @@
 #include "reset_domain.h"
 #include "voltage_domain.h"
 
+#define SCMI_PLAYLOAD_U32_MAX		(SCMI_PLAYLOAD_MAX / sizeof(uint32_t))
+
+/* Provision input message payload buffers for each supported entry */
+static uint32_t fast_smc_payload[CFG_TEE_CORE_NB_CORE][SCMI_PLAYLOAD_U32_MAX]
+__maybe_unused;
+
+static uint32_t interrupt_payload[CFG_TEE_CORE_NB_CORE][SCMI_PLAYLOAD_U32_MAX]
+__maybe_unused;
+
+static uint32_t threaded_payload[CFG_NUM_THREADS][SCMI_PLAYLOAD_U32_MAX]
+__maybe_unused;
+
+/* SMP protection on channel access */
+static unsigned int smt_channels_lock;
+
+/* If channel is not busy, set busy and return true, otherwise return false */
+bool scmi_channel_set_busy(struct scmi_msg_channel *channel)
```

Comment:
I would have named this `scmi_channel_try_set_busy()`.
Or perhaps `scmi_try_claim_channel()` to be paired with `scmi_release_channel()`.